ENH: Allow sending notifications to everyone but oneself via notify-send-others

inkarkat opened the pull request:
Note: This goes on top of / builds upon #1

inkarkat committed:
ENH: Allow sending notifications to everyone but oneself via notify-send-others

If I'm doing things like shutting down the system, I don't need a notification for myself; only other users need to be informed.
With this, if no other user is logged in, the command becomes a no-op.

inkarkat commented:
FWIW, I quickly found use cases to invoke commands other than `notify-send` for all desktop users, and decided to implement a separate [eachDesktopUserDo](https://github.com/inkarkat/desktop/blob/master/bin/eachDesktopUserDo) command, and then implement [notify-send-all](https://github.com/inkarkat/desktop/blob/master/bin/notify-send-all), [notify-send-others](https://github.com/inkarkat/desktop/blob/master/bin/notify-send-others), and [notify-send-to](https://github.com/inkarkat/desktop/blob/master/bin/notify-send-to) in terms of that (all as separate scripts; I don't like changing behavior based on the command's name (though it is a nifty trick)).

hackerb9 commented on notify-send-all:
```diff
@@ -120,6 +123,9 @@ main() {
 	    ;;
+	notify-send-others)
+	    skipuser="${USER:?}"
+	    ;&
```
Why die if if `$USER` is not set? There are cases where that can happen and it needn't be fatal. Maybe something like, ${USER:-`whoami`}.

Also, what's up with line 128? It's a simple fix, but now I'm wondering if you accidentally sent the untested version of your patch.

inkarkat replied:
Those cases are few, but I've added fallbacks as it's easy to do.

The `;&` is a fall-through to the next branch of the switch-statement, to avoid repeating the `send-all` command.

hackerb9 replied:
Learn something new every day. I didn't know bash case statements could fall through. Thanks.

hackerb9 commented:
I like the idea of not notifying oneself and maybe it should just be the default.

inkarkat committed:
Robustness: Implement fallback if $USER is unset

For the given use cases, this is unlikely (as login sets $USER), but there are easy alternatives.
Source: https://unix.stackexchange.com/questions/76354/who-sets-user-and-username-environment-variables

hackerb9 reviewed (approved):
Looks good now
